### PR TITLE
Actually detect negative sizes by following ES.106

### PR DESCRIPTION
**Pull request by Artikash**
And don't senselessly use one letter names.

### CODE REVIEW

File: CppCoreGuidelines.md [hunk 1]
```diff
@@ -20320,20 +20320,20 @@ and errors (when we didn't deal correctly with semi-constructed objects consiste
...
         {
             // invariant check
-            Expects(s > 0);
-            return s;
+            Expects(size > 0);
+            return size;
         }
 
     public:
         // even more better would be a class for a 2D Size as one single parameter
```
Comment (johelegp) [inline CppCoreGuidelines.md]:
```suggestion
        // even better would be a class for a 2D Size as one single parameter
```

File: CppCoreGuidelines.md [hunk 2]
```diff
@@ -20320,20 +20320,20 @@ and errors (when we didn't deal correctly with semi-constructed objects consiste
 
     class Picture
     {
-        size_t mx;
-        size_t my;
+        ptrdiff_t mx;
+        ptrdiff_t my;
```
Comment (cubbimew) [inline CppCoreGuidelines.md]: why not just `int` then?

Comment (Artikash) [reply CppCoreGuidelines.md]: To handle really large sizes appropriately. I guess it should be consistent though, not sure whether to change the above to prrdiff_t or this to int

Comment (markyin) [reply CppCoreGuidelines.md]: `ptrdiff_t` is less than `size_t`.

https://en.cppreference.com/w/cpp/types/ptrdiff_t

**Comment by hsutter** [comment]
Editors call: We agree the members should be signed, we'll merge the PR and then change them to `int` just to mirror the "bad" example. Thanks!

**Pull request merged**